lnwire/funding_locked.go

```diff
@@ -0,0 +1,139 @@
+	// the next commitment transaction for the channel.
+	NextPerCommitmentPoint *btcec.PublicKey
+}
+
+// NewFundingLocked creates a new FundingLocked message, populating it with
+// the necessary IDs and revocation secret..
+func NewFundingLocked(op *wire.OutPoint, cid ChannelID, ans, abs *btcec.Signature,
+	npcp *btcec.PublicKey) *FundingLocked {
+	return &FundingLocked{
+		ChannelOutpoint:        op,
+		ChannelID:              cid,
+		NextPerCommitmentPoint: npcp,
+	}
+}
+
+// A compile time check to ensure FundingLocked implements the
+// lnwire.Message interface.
+var _ Message = (*FundingLocked)(nil)
+
+// Decode deserializes the serialized FundingLocked message stored in the passed
+// io.Reader into the target FundingLocked using the deserialization
+// rules defined by the passed protocol version.
+//
+// This is part of the lnwire.Message interface.
+func (c *FundingLocked) Decode(r io.Reader, pver uint32) error {
+	err := readElements(r,
+		&c.ChannelOutpoint,
+		&c.ChannelID,
+		&c.NextPerCommitmentPoint)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Encode serializes the target FundingLocked message into the passed io.Writer
+// implementation. Serialization will observe the rules defined by the passed
+// protocol version.
+//
+// This is part of the lnwire.Message interface.
+func (c *FundingLocked) Encode(w io.Writer, pver uint32) error {
+	err := writeElements(w,
+		c.ChannelOutpoint,
+		c.ChannelID,
+		c.NextPerCommitmentPoint)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Command returns the uint32 code which uniquely identifies this message as a
+// FundingLocked message on the wire.
+//
+// This is part of the lnwire.Message interface.
+func (c *FundingLocked) Command() uint32 {
+	return CmdFundingLocked
+}
+
+// MaxPayloadLength returns the maximum allowed payload length for a
+// FundingLocked message. This is calculated by summing the max length of all
+// the fields within a FundingLocked message.
+//
+// This is part of the lnwire.Message interface.
+func (c *FundingLocked) MaxPayloadLength(uint32) uint32 {
+	var length uint32
+
+	// ChannelOutpoint - 36 bytes
+	length += 36
+
+	// ChannelID - 8 bytes
+	length += 8
+
+	// NextPerCommitmentPoint - 33 bytes
+	length += 33
+
+	return length
+}
+
+// Validate examines each populated field within the FundingLocked message for
+// field sanity. For example, signature fields MUST NOT be nil.
+//
+// This is part of the lnwire.Message interface.
+func (c *FundingLocked) Validate() error {
+	if c.ChannelOutpoint == nil {
+		return fmt.Errorf("the funding outpoint must be non-nil.")
+	}
+
+	if c.NextPerCommitmentPoint == nil {
+		return fmt.Errorf("The next per commitment point must be non-nil.")
+	}
+
+	// We're good!
+	return nil
+}
+
+// String returns the string representation of the FundingLocked message.
+//
+// This is part of the lnwire.Message interface.
+func (c *FundingLocked) String() string {
```

Comment:
Recently all `String()` methods in the `lnwire` package have been removed in favor of the more informative output generated by `spew.Sdump`. 

However, to avoid the logs getting a bit spammy, you'll also need to add an entry for `FundingLocked` in this function: https://github.com/lightningnetwork/lnd/blob/master/peer.go#L526